pkg/middleware/trace.go

```diff
@@ -0,0 +1,36 @@
+/****************************************************************************
+ * Copyright 2023 Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package middleware //
+package middleware
+
+import (
+	"net/http"
+
+	"go.opentelemetry.io/otel"
+)
+
+func AddTracing(tracerName, spanName string) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			ctx, span := otel.Tracer(tracerName).Start(r.Context(), spanName)
+			defer span.End()
```

Comment:
This middleware solution looks good. Wondering any way we can add attributes consistently for all APIs.

Reply:
collecting attributes from middleware would be difficult, I think. Also in some API, it seems there is hardly any attributes (extra info) to collect.

Reply:
btw added some common http attributed in the middleware

Reply:
Most of the APIs don't have it. Also it seems those attributes are just considered as extra information in tracing sounds like less important things (not in our case btw :) ). Maybe we can leave it for now.